xml/issue3183.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version='1.0' encoding='utf-8' standalone='no'?>
+<!DOCTYPE issue SYSTEM "lwg-issue.dtd">
+
+<issue num="3183" status="New">
+<title>Normative permission to specialize Ranges variable templates</title>
+<section><sref ref="[iterator.concept.sizedsentinel]"/></section>
+<submitter>Casey Carter</submitter>
+<date>14 Jan 2019</date>
+<priority>99</priority>
+
+<discussion>
+<p>
+<a href="https://wg21.link/p0896r4">P0896R4 "The One Ranges Proposal"</a> added
+boolean variable templates <tt>std::disable_sized_sentinel</tt> and
+<tt>std::ranges::disable_sized_range</tt> which users are intended to specialize
+to <tt>false</tt> for program-defined <tt>Iterator</tt>-<tt>Sentinel</tt> pairs
+/ <tt>Range</tt> types which meet the syntax but do not model
+the semantics of the <tt>SizedSentinel</tt> / <tt>SizedRange</tt> concepts,
+respectively. Specializing these traits allows the use of such types with the
+library which would otherwise treat them as if they model <tt>SizedSentinel</tt>
+/ <tt>SizedRange</tt>.
+</p><p>
+The wording in P0896R4 failed, however, to provide normative permission to
+specialize these variable templates as is required by <sref ref="[namespace.std]"/>
+after the application of <a href="https://wg21.link/p0551r3">P0551R3</a>.
+</p>
+</discussion>
+
+<resolution>
+<p>This wording is relative to <a href="https://wg21.link/n4791">N4791</a>.</p>
+
+<ol>
+<li><p>Change <sref ref="[iterator.concept.sizedsentinel]"/> as follows:</p>
+<blockquote>
+<p>
+[&hellip;]
+</p><p>
+(2.2) &mdash; If <tt>−N</tt> is representable by
+<tt>iter_difference_t&lt;I&gt;</tt>, then <tt>i - s</tt> is well-defined and
+equals <tt>−N</tt>.
+</p><p>
+<ins>-?- Pursuant to <sref ref="[namespace.std]"/>, users may specialize
+<tt>disable_sized_sentinel</tt> for cv-unqualified types <tt>S</tt> and
```

Comment:
Why disallow cv-qualification but allow references?

Reply:
No good reason. Do you see any holes in "cv-unqualified non-array object types"?

Reply:
OK, but should we include pointers too? Do we want to allow specializing `disable_sized_sentinel<MyType*, MyType*>` to true?

Reply:
No, but we *do* want to allow `disable_sized_sentinel<int*, MySentinel>` and `disable_sized_sentinel<MyIterator, int*>`. 

"cv-unqualified non-array object types, at least one of which depends on a program-defined type, and at least one of which is not a pointer type"?

Reply:
"cv-unqualified non-array object types, at least one of which is a program-defined type"?